src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -1143,6 +1143,28 @@ protected function parseNestedWith($name, $results)
     }
 
     /**
+     * Add the given scopes to the current builder instance.
+     *
+     * @param  array  $scopes
+     * @return mixed
+     */
+    public function scopes(array $scopes)
+    {
+        $builder = $this;
+
+        foreach ($scopes as $scope => $parameters) {
+            if (is_int($scope)) {
+                $scope = $parameters;
+                $parameters = [];
+            }
+
+            $builder = $builder->callScope('scope'.ucfirst($scope), (array) $parameters);
```

Comment:
@sileence This line doesn't work. ``callScope`` needs ``callable`` type as the first argument, so it should be:

```php
$callable = ['scope'.ucfirst($scope), $this->model];

$builder->callScope($callable, (array) $parameters)
```

Also, scope can return anything, not necessarily a builder instance, so I wouldn't re-assign it here.